### ISSUE TITLE
Broken Focus With Unmapped Windows

### ISSUE THREAD

**Issue by adamnew123456** [adamnew123456/SmallWM#6]
# Steps To Reproduce

1. Open Evince (Evince in particular, because of how it treats its About dialogue under the hood)
2. Click the "About" button.
3. Close the "About" dialogue.
4. Note that SmallWM loses all keyboard focus - focus cycling does not work, nor does switching desktops, or any of the other shortcuts.

# Background

Evince doesn't like having more than one "About" dialogue - it is allocated once, and then re-used every time the user clicks "About." I have no idea why you'd do something like this, but they do. You can prove this to yourself by using `xwininfo` and nothing that the *MapState* is *IsUnMapped* when the dialogue is closed.

# Proposed Fix

Add a handler to the event loop, which detects unmapping and arranges to remove the focus from the unmapped window.

**Comment by adamnew123456** [comment]
Despite fixing this in an unpushed commit, Evince in particular also has the issue that the window being unmapped will remain in the focus cycle, which somehow causes things to break in an unexpected way.

**Comment by adamnew123456** [comment]
6ce575d fixes this bug.
